src/core/services/aggregation.js

```diff
@@ -1,34 +1,66 @@
+import * as _ from './utility';
+
 export default class Aggregation {
 	constructor() {
 	}
 
 	static first(rows, getValue) {
-		if(!rows.length){
+		if (!rows.length) {
 			return null;
 		}
 
 		return getValue(rows[0]);
 	}
 
 	static last(rows, getValue) {
-		if(!rows.length){
+		if (!rows.length) {
 			return null;
 		}
 
 		return getValue(rows[rows.length - 1]);
 	}
 
 	static max(rows, getValue) {
-		if(!rows.length){
+		if (!rows.length) {
 			return null;
 		}
 
-		var max = Math.max.apply(Math, rows.map(getValue));
-		return max;
+		return getValue(_.maxBy(rows, d => getValue(d)));
+	}
```

Comment:
1. _.maxBy(rows, getValue) - is enough
2. Let's write our own super fast version with while(true), it's easy and agg perfomance is critical operation for grouping aggregations